### PR TITLE
global: unify build and deployment script

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: build-and-deploy.sh [hunk 1]
```diff
@@ -6,9 +6,10 @@
 # inspirehep is free software; you can redistribute it and/or modify it under
 # the terms of the MIT License; see LICENSE file for more details.
 
-IMAGE="inspirehep/ui"
-DOCKER_CONTEXT='ui'
-TAG="$(git describe --always)"
+TAG="${TRAVIS_TAG:-$(git describe --always)}"
```
Comment (michamos) [inline build-and-deploy.sh]:
```suggestion
TAG="${TRAVIS_TAG:-$(git describe --always --tags)}"
```
in order to allow lightweight (non-annotated) tags too, which are the type of tags generated by github releases.

Comment (drjova) [reply build-and-deploy.sh]: 👍 TRAVIS_TAG will be set though

File: build-and-deploy.sh [hunk 2]
```diff
@@ -21,18 +22,30 @@ login() {
...
+    -t "${IMAGE_BACKEND}" \
+    backend
 
-  echo "Pushing image to ${IMAGE}:${TAG}"
-  retry docker push "${IMAGE}:${TAG}"
-  retry docker push "${IMAGE}"
+  echo "Pushing image to ${IMAGE_BACKEND}:${TAG}"
+  retry docker push "${IMAGE_BACKEND}:${TAG}"
+  retry docker push "${IMAGE_BACKEND}"
 }
```
Comment (benjamin-bergia) [inline build-and-deploy.sh]: I would do this instead
```bash
buildPush() {
  context="${1}"
  image="${2}"

  echo "Building docker image for ${context}"
  retry docker build \
    --build-arg VERSION="${TAG}" \
    -t "${image}:${TAG}" \
    -t "${image}" \
    ${context}

  echo "Pushing image to ${image}:${TAG}"
  retry docker push "${image}:${TAG}"
  retry docker push "${image}"
}
```

File: build-and-deploy.sh [hunk 3]
```diff
@@ -53,7 +66,8 @@ deployQA() {
 
 main() {
   login
-  buildPush
+  buildPushUI
+  buildPushBackend
```
Comment (benjamin-bergia) [inline build-and-deploy.sh]: And then here
```bash
buildPush "ui" "inspirehep/ui"
buildPush "backend" "inspirehep/backend"
```

**Pull request merged**